File: Codigos en Java/Ejercicio8_Ape06.java
import java.util.ArrayList; 
public class Ejercicio8_Ape06 { 
public static void main(String[] args) { 
final int N = 100000; 
// Inserción en arreglo 
int[] arreglo = new int[N]; 
long t1 = System.nanoTime(); 
for (int i = 0; i < N; i++) arreglo[i] = i; 
long t2 = System.nanoTime(); 
// Inserción en ArrayList 
ArrayList<Integer> lista = new ArrayList<>(); 
long t3 = System.nanoTime(); 
for (int i = 0; i < N; i++) lista.add(i); 
long t4 = System.nanoTime(); 
System.out.println("Insercion de " + N + " elementos:"); 
System.out.println("Arreglo  : " + (t2-t1)/1000 + " microsegundos"); 
System.out.println("ArrayList: " + (t4-t3)/1000 + " microsegundos"); 
} 
}

File: Codigos en C++/Ejercicio8_Ape06.cpp
#include <iostream> 
#include <vector> 
#include <chrono> 
using namespace std; 
using namespace std::chrono; 
int main() { 
const int N = 100000; 
// Inserción en arreglo (simulada con índice) 
int arreglo[N]; 
auto t1 = high_resolution_clock::now(); 
for (int i = 0; i < N; i++) arreglo[i] = i; 
auto t2 = high_resolution_clock::now(); 
auto durArr = duration_cast<microseconds>(t2 - t1).count(); 
// Inserción en vector 
vector<int> vec; 
auto t3 = high_resolution_clock::now(); 
for (int i = 0; i < N; i++) vec.push_back(i); 
auto t4 = high_resolution_clock::now(); 
auto durVec = duration_cast<microseconds>(t4 - t3).count(); 
cout << "Insercion " << N << " elementos:" << endl; 
cout << "Arreglo: " << durArr << " microsegundos" << endl; 
cout << "Vector : " << durVec << " microsegundos" << endl; 
return 0; 
}